spec/index.html

```diff
@@ -1493,3 +1503,221 @@ <h1>SharedArray ( ..._values_ )</h1>
+            1. Set _done_ to *true*.
+        1. Perform LeaveCriticalSection(_WL_).
+        1. If _waiterRecord_.[[Result]] is *"ok"*, then
+          1. Return ~acquired~.
+        1. Assert: _waiterRecord_.[[Result]] is *"timed-out"*.
+        1. Return ~timed-out~.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-unlockmutex" type="abstract operation">
+      <h1>
+        UnlockMutex (
+          _mutex_: an Object,
+        ) : ~unused~
+      </h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. Assert: _mutex_ has a [[MutexWaiterList]] internal slot.
+        1. Let _WL_ be _mutex_.[[MutexWaiterList]].
+        1. Perform EnterCriticalSection(_WL_).
+        1. Assert: _mutex_.[[IsLocked]] is *true*.
+        1. Set _mutex_.[[IsLocked]] to *false*.
+        1. Let _S_ be RemoveWaiters(_WL_, 1).
+        1. For each element _W_ of _S_, do
+          1. Perform NotifyWaiter(_WL_, _W_).
+        1. Perform LeaveCriticalSection(_WL_).
+        1. Return ~unused~.
+      </emu-alg>
+    </emu-clause>
+  </emu-clause>
+
+  <emu-clause id="sec-mutex-object">
+    <h1>The Mutex Constructor</h1>
+    <p>The Mutex constructor:</p>
+    <ul>
+      <li>is <dfn>%Mutex%</dfn>.</li>
+      <li>is the initial value of the *"Mutex"* property of the %Atomics% object.</li>
+      <li>creates and initializes a new Mutex when called as constructor.</li>
+      <li>is not intended to be called as a function and will throw an exception when called in that manner.</li>
+    </ul>
+
+    <emu-clause id="sec-atomics.mutex">
+      <h1>Atomics.Mutex ( )</h1>
+      <p>This function performs the following steps when called:</p>
+      <emu-alg>
+        1. If NewTarget is *undefined*, throw a *TypeError* exception.
+        1. Let _createInitializer_ be a new Abstract Closure with parameters (_newMutex_) that captures nothing and performs the following steps when called:
+          1. Set _newMutex_.[[MutexWaiterList]] to a new WaiterList Record.
+          1. Set _newMutex_.[[IsLocked]] to *false*.
+          1. Return ~unused~.
+        1. Return SharedStructCreate(_createInitializer_, « [[MutexWaiterList]], [[IsLocked]] »).
+      </emu-alg>
+    </emu-clause>
+  </emu-clause>
+
+  <emu-clause id="sec-properties-of-the-mutex-constructor">
+    <h1>Properties of the Mutex Constructor</h1>
+    <p>The Mutex constructor:</p>
+    <ul>
+      <li>has a [[Prototype]] internal slot whose value is %Function.prototype%.</li>
+      <li>has the following properties:</li>
+    </ul>
+
+    <emu-note>
+      <p>Per-Realm prototypes, which is currently an open design question and not included in this draft, will give Mutexes a per-Realm prototype with built-in methods instead of static methods, and will also add support for the %Symbol.dispose%.</p>
+    </emu-note>
+
+    <emu-clause id="sec-atomics.mutex.lock">
+      <h1>Atomics.Mutex.lock ( _mutex_ [ , _unlockToken_ [ , _timeout_ ] ] )</h1>
```

Comment:
Also wait why is this static instead of prototype?

Reply:
Because shared structs can't point to unshared objects, and functions can't be shared, so having a prototype requires something like per-Realm prototypes that I have talked about in the past, together with solving the "correlation problem" (the registry thing). We're still actively discussing how to solve this issue with Mark & co, and have decided to leave it as an open design question to be settled during Stage 2. As such, the extra complexity of per-Realm prototypes is omitted from the spec draft.

If per-Realm prototypes get consensus, then this will definitely be a prototype.

Reply:
(It's my interpretation that where these methods live isn't as major a semantics point as what they do.)